### PR TITLE
feat(manager/gradle): support short dependency notation for named Kotlin modules

### LINKED ISSUE #19147: Support "kotlin(...)" artifact coordinate builder
### What would you like Renovate to be able to do?

In the recent PR from Renovate, it wasn't able to detect this dependency:

```kotlin
implementation(platform(kotlin("bom", "1.7.21")))
```

In fact it's a shortcut for
```kotlin
implementation(platform("org.jetbrains.kotlin:kotlin-bom:1.7.21"))
```

To mitigate the issue, I had to make this change: https://github.com/krzema12/github-workflows-kt/pull/574/files but ideally Renovate should recognize this fairly popular shortcut.

### If you have any ideas on how this should be implemented, please tell us here.

Probably tinkering around regexes that are currently in place for Gradle.

### Is this a feature you are interested in implementing yourself?

Maybe

### PR DESCRIPTION
**Pull request by Churro**
## Changes

Support for dependency declarations using `kotlin(...)`, an extension function that [Gradle ships natively](https://github.com/gradle/gradle/blob/efe4278c15960007056d02720d0d66d3e184a248/build-logic/kotlin-dsl/src/main/kotlin/gradlebuild/kotlindsl/generator/tasks/GenerateKotlinDependencyExtensions.kt#L85-L92) with kotlin-dsl.

This PR enables the following flavors (+ parameterized string values):
```kotlin
kotlin("bom", "1.2.3")
kotlin("bom", version = "1.2.3")
```

This implementation requires two parameters, whereas the `kotlin(...)` used with plugins requires just one. In other words, no side effects to be expected.

_Note_: The handler code follows the same logic as `handleLongFormDep`. I intend to extract such commonalities in a future refactoring PR.

## Context

* Closes #19147
 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

Test repository (= repo linked in #19147 with HEAD~2): 
* https://github.com/renovate-demo/19147-github-workflows-kt/pull/1

<details><summary>Logs</summary>

```
"gradle": [
  ...
  {
    "packageFile": "buildSrc/build.gradle.kts",
    "datasource": "maven",
    "deps": [
    ...
     {
       "depName": "bom",
       "currentValue": "1.7.21",
       "packageName": "org.jetbrains.kotlin:kotlin-bom",
       "managerData": {
         "fileReplacePosition": 192,
         "packageFile": "buildSrc/build.gradle.kts"
       },
       "fileReplacePosition": 192,
       "registryUrls": [
         "https://repo.maven.apache.org/maven2",
         "https://plugins.gradle.org/m2/"
       ],
       "depType": "devDependencies",
       "depIndex": 1,
       "updates": [
         {
           "bucket": "non-major",
           "newVersion": "1.7.22",
           "newValue": "1.7.22",
           "releaseTimestamp": "2022-11-28T09:54:08.000Z",
           "newMajor": 1,
           "newMinor": 7,
           "updateType": "patch",
           "branchName": "renovate/kotlin-monorepo"
         }
       ],
       "warnings": [],
       "versioning": "gradle",
       "sourceUrl": "https://github.com/JetBrains/kotlin",
       "homepage": "https://kotlinlang.org/",
       "currentVersion": "1.7.21",
       "isSingleVersion": true,
       "fixedVersion": "1.7.21"
     },
    ...
    ]
  }
],
```

</details>

### CODE REVIEW

**Review by zharinov — APPROVED** [review]
LGTM

**Review by rarkins — APPROVED** [review]

**Comment by Churro** [comment]
@rarkins, would it be already possible to merge this or is something still pending?

File: lib/modules/manager/gradle/parser/handlers.ts [hunk 1]
```diff
@@ -110,6 +110,48 @@ export function handleDepInterpolation(ctx: Ctx): Ctx {
...
+  const moduleNameTokens = loadFromTokenMap(ctx, 'moduleName');
+  const versionTokens = loadFromTokenMap(ctx, 'version');
+
+  const moduleName = interpolateString(moduleNameTokens, ctx.globalVars);
+  const versionValue = interpolateString(versionTokens, ctx.globalVars);
+  if (!moduleName || !versionValue) {
+    return ctx;
+  }
+
+  const groupIdArtifactId = `org.jetbrains.kotlin:kotlin-${moduleName}`;
```
Comment (viceice) [inline lib/modules/manager/gradle/parser/handlers.ts:123]: can we inline this?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Review by viceice — APPROVED** [review]

**Pull request merged**